modules/storage/storage-account/main.bicep

```diff
@@ -55,6 +151,30 @@ resource storageAccount 'Microsoft.Storage/storageAccounts@2022-09-01' = {
   }
 }
 
+resource blobService 'Microsoft.Storage/storageAccounts/blobServices@2022-09-01' = if (enableBlobService) {
+  name: 'default'
+  parent: storageAccount
+  properties: blobProperties
+}
+
+resource container 'Microsoft.Storage/storageAccounts/blobServices/containers@2022-09-01' = if (enableBlobService) {
+  name: blobContainerName
+  parent: blobService
+  properties: blobContainerProperties
+}
+
+module storageRbac '.bicep/nested_rbac.bicep' = [for (roleAssignment, index) in roleAssignments: {
+  name: '${uniqueString(deployment().name, location)}-acr-rbac-${index}'
+  params: {
```

Comment:
Hi @dciborow 
So, I updated and tested all other changes you suggested. Then I converted roleAssignments to a custom type. But I am not sure how we handle the parsing logic (which involves a loop) if we pass this as param to nested bicep (which already runs another loop). Because nested loops are something not supported in bicep natively as far as I searched through (https://github.com/Azure/bicep/issues/2013), it doesn't seem straight forward. 
Another thing is I need to repeat the custom type inside the nested template as well which I don't find intuitive. Do you suggest how do we achieve that?